include/MiniLua/values.hpp

```diff
@@ -323,7 +324,7 @@ class Number {
     [[nodiscard]] auto is_int() const -> bool;
     [[nodiscard]] auto is_float() const -> bool;
 
-    template <typename Fn> auto visit(Fn fn) const {
+    template <typename Fn> [[nodiscard]] auto visit(Fn fn) const {
```

Comment:
I'm not sure if `nodiscard` is correct here because you might not always want to return something from the visit. But I think nodicard is only a warning so it probably doesn't matter.

Reply:
if there is no value returned in the visit, there is no return in visit if i understand it correctly.
and if there is a return in the visit the value should not be thrown away in my opinion. Also this change was a suggestion of clang and I found it reasonable with the thoughts mentioned above.

Reply:
I don't really know how it interacts if there is no return. You are still returning void. Maybe nodiscard is ignored if you return void.